lib/storage/storage_test.go

```diff
@@ -156,6 +157,30 @@ func testDateMetricIDCache(c *dateMetricIDCache, concurrent bool) error {
 	return nil
 }
 
+func TestDateMetricIDCacheIsConsistent(_ *testing.T) {
+	const (
+		generation  = 1
+		date        = 1
+		concurrency = 2
+		numMetrics  = 100000
+	)
+	dmc := newDateMetricIDCache()
+	var wg sync.WaitGroup
+	for i := range concurrency {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for id := uint64(i * numMetrics); id < uint64((i+1)*numMetrics); id++ {
+				dmc.Set(generation, date, id)
+				if !dmc.Has(generation, date, id) {
+					panic(fmt.Errorf("dmc.Has(metricID=%d): unexpected cache miss after adding the entry to cache", id))
```

Comment:
Could you please replace panic with `t.Fatalf` ? It requires to change function signature `(t *testing.T)`

Reply:
Please allow me to explain before I change anything. So `t.Fatalf` shouldn't be called from goroutines and, when running `make check-all`, `go vet` would complain (even if I add that param to the goroutine):

```
go vet ./lib/...
# github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
# [github.com/VictoriaMetrics/VictoriaMetrics/lib/storage]
lib/storage/storage_test.go:176:6: call to (*testing.T).Fatalf from a non-test goroutine
```

Instead, the error would need to be propataged to the main test goroutine which'd complicate the code. And using `panic` somewhat simplifies it.

Let me know what you think. Thanks!